tests/test_cte.py

```diff
@@ -711,3 +713,132 @@ def test_django52_ambiguous_column_names(self):
+            qs = with_cte(cte, select=cte)
+            self.assertEqual(list(qs), [
+                ('earth', 30),
+                ('earth', 31),
+                ('earth', 32),
+                ('earth', 33),
+                ('mars', 40),
+                ('mars', 41),
+                ('mars', 42),
+                ('mercury', 10),
+                ('mercury', 11),
+                ('mercury', 12),
+                ('moon', 1),
+                ('moon', 2),
+                ('moon', 3),
+                ('proxima centauri', 2000),
+                ('proxima centauri b', 10),
+                ('proxima centauri b', 11),
+                ('proxima centauri b', 12),
+                ('sun', 1000),
+                ('venus', 20),
+                ('venus', 21),
+                ('venus', 22),
+                ('venus', 23),
+            ])
+
+        def test_queryset_after_values_list_flat(self):
+            cte = CTE(
+                Order.objects.values_list("region", flat=True)
+                .order_by("region")
+                .distinct()
+            )
+            qs = with_cte(cte, select=cte)
+            self.assertEqual(list(qs), [
+                'earth',
+                'mars',
+                'mercury',
+                'moon',
+                'proxima centauri',
+                'proxima centauri b',
+                'sun',
+                'venus'
+            ])
+
+        def test_queryset_values_list_order1(self):
+            cte = CTE(
+                Order.objects.values("region")
+                .annotate(c=Count("region"))
+                .values_list("c", "region")
+                .order_by("region")
+            )
+            qs = with_cte(cte, select=cte)
+            # Use the `ValuesListIterable` for being able to inspect the order of the fields
+            # in the resulting query
+            qs._iterable_class = ValuesListIterable
+            # Ensure the order of queried fields is the specified one: c, region
+            # Before the fix, the order would have been this one: region, c
+            self.assertEqual(list(qs), [
+                (4, 'earth'),
+                (3, 'mars'),
+                (3, 'mercury'),
+                (3, 'moon'),
+                (1, 'proxima centauri'),
+                (3, 'proxima centauri b'),
+                (1, 'sun'),
+                (4, 'venus'),
+            ])
+
+        def test_queryset_values_list_order2(self):
+            cte = CTE(
+                Order.objects.values("region")
+                .annotate(r=F("region"), c=Count("region"))
+                .values_list("c", "r")
+                .order_by("r")
+            )
+            qs = with_cte(cte, select=cte)
+            # Use the `ValuesListIterable` for being able to inspect the order of the fields
+            # in the resulting query
+            qs._iterable_class = ValuesListIterable
```

Comment:
Same as previous comment, this test passes without the assignment.

```suggestion
```

Alternately, maybe you prefer an assertion?

```suggestion
            assert qs._iterable_class is ValuesListIterable
```